esyi/Sandbox.ml

```diff
@@ -121,40 +133,46 @@ let ofDir ~cfg (path : Path.t) =
       Manifest.toPackage ~version manifest
     in
 
+   let%bind origin = originOfPath path in
+
     return {
       cfg;
       path;
       root;
       resolutions;
       ocamlReq;
       dependencies = Package.Dependencies.NpmFormula reqs;
+      origin;
     }
   | None ->
     begin match%bind readAggregatedOpamManifest path with
-    | Some (dependencies, devDependencies) ->
-
-      let root = {
-        Package.
-        name = "root";
-        version = Package.Version.Source (Package.Source.LocalPath path);
-        source = Package.Source (Package.Source.LocalPath path), [];
-        dependencies = Package.Dependencies.OpamFormula dependencies;
-        devDependencies = Package.Dependencies.OpamFormula devDependencies;
-        opam = None;
-        kind = Package.Esy;
-      } in
-
-      let dependencies =
-        Package.Dependencies.OpamFormula (dependencies @ devDependencies)
-      in
+      | Some (dependencies, devDependencies) ->
+
+        let root = {
+          Package.
+          name = "root";
+          version = Package.Version.Source (Package.Source.LocalPath path);
+          source = Package.Source (Package.Source.LocalPath path), [];
+          dependencies = Package.Dependencies.OpamFormula dependencies;
+          devDependencies = Package.Dependencies.OpamFormula devDependencies;
+          opam = None;
+          kind = Package.Esy;
+        } in
+
+        let dependencies =
+          Package.Dependencies.OpamFormula (dependencies @ devDependencies)
+        in
 
-      return {
-        cfg;
-        path;
-        root;
-        resolutions = Manifest.Resolutions.empty;
-        dependencies;
-        ocamlReq = Some ocamlReqAny;
-      }
-    | None -> error "unable to find either package.json or opam files"
+        let%bind origin = originOfPath path in
```

Comment:
I was able to get rid of the other call for the npm case by returning the origin from `readPackageJsonManifest`. Still thinking about how to handle the Opam case since there may not be a single file. If it is one file then it's whatever that filename is, otherwise it's a `Path.t list`. Maybe `readAggregatedOpamManifest` could be similarly altered to return an `origin` depending on how many files it found and we could add back the `AggregatedOpam` case. For the moment I removed it.